### DISCUSSION TITLE
sponsored_products Reports BUG

### DISCUSSION THREAD

**Discussion by pengpengker** [denisneuf/python-amazon-ad-api#66] [General]
**Describe the bug**
1-bug: reports.py-41  kwargs.pop('body')  is error,Nofound body in kwargs.
2-bug: client.py-239   Content-type is incorrect, resulting in a return status of 400/422, params should not use the passed data.In general there are multiple errors here

sponsored_products Reports  post_report
Not submitting the request with the Content-type will result 422/400 status_code。

**To Reproduce**
1. call sponsored_products.Reports(credentials=my_credentials, marketplace=Marketplaces.NA, debug=True).post_report(
    recordType="campaigns",
    metrics='impressions',
    reportDate="20220729"
)

**Additional context**
I'll submit new code when I'm done, if you're willing to accept。 grateful！

**Comment by denisneuf** [comment]
Check the help:

https://python-amazon-ad-api.readthedocs.io/en/latest/sp/reports.html

You are using wrong you need to pass your values as a dictionary:

from ad_api.api.sp.reports import Reports

file = open("ad_groups.json")
data = file.read()
file.close()

record_type = 'adGroups'

result = Reports().post_report(
    recordType=record_type,
    **body=data**
)

payload = result.payload
report_id = payload.get('reportId')

While the dictionary should look like this:

{
  "stateFilter": "enabled",
  "reportDate": "20210917",
  "metrics": "campaignName,campaignId,adGroupName,adGroupId,impressions,clicks,attributedConversions30d"
}

**Reply by pengpengker** [reply]
Get it. I was misled by my usage habits. sorry, i'll take care of that
Thanks for your excellent work！
